client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -677,6 +677,29 @@ Undocumented.prototype.getDomainContactInformation = function( fn ) {
 	);
 };
 
+/**
+ * Retrieves the contact validation schema for the specified tld
+ *
+ * @param {string} tld The domain tld to validate for.
+ * @param {Function} fn The callback function
+ * @return {Promise} A Promise to resolve when complete
+ * @api public
+ */
+Undocumented.prototype.getDomainContactInformationValidationSchema = function( tld, fn ) {
+	debug( '/domains/validation-schema/ query' );
+
+	return this.wpcom.req.get( { path: '/domains/validation-schema/' + tld }, function(
+		error,
+		data
+	) {
+		if ( error ) {
+			return fn( error );
+		}
+
+		return fn( null, data );
+	} );
+};
```

Comment:
some of us really don't like `undocumented` and we're not adding functions here anymore as a project either. for now it's probably even less work overall to just inline the `wpcom.js` call. we have moved to the data layer but I think in an exploratory PR like this it's okay to doodle with `wpcom.js` - maybe best not to merge it into **master** but it may support the exploration.

Reply:
Good call. I think a data component to fetch and own the schema is a better approach.

I've been wondering  about making that different component - separating out the fetching and validation concerns - but I'm not sure if it's over-engineering.